src/main/java/com/uid2/admin/audit/QLDBAuditMiddleware.java

```diff
@@ -0,0 +1,134 @@
+package com.uid2.admin.audit;
+
+import com.amazon.ion.IonStruct;
+import com.amazon.ion.IonSystem;
+import com.amazon.ion.IonValue;
+import com.amazon.ion.system.IonSystemBuilder;
+import com.uid2.admin.auth.AdminUserProvider;
+import com.uid2.shared.auth.IAuthorizable;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.web.RoutingContext;
+import software.amazon.awssdk.services.qldbsession.QldbSessionClient;
+import software.amazon.qldb.QldbDriver;
+import software.amazon.qldb.RetryPolicy;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class QLDBAuditMiddleware implements AuditMiddleware{
```

Comment:
From design perspective, the logic of middleware which collects audit information should be separate from logic of storing the audit record(s) to the destination log/db.

Reply:
Working on splitting logic